### ISSUE TITLE
About experiment with pseudo label

### ISSUE THREAD

**Issue by KhaLee2307** [ku21fan/STR-Fewer-Labels#17]
Thank you for your gorgeous research. I have some questions about your experiment (focus on  Semi-Supervised Learning):
1. When you use pseudo labels to retrain your model, do you select all of them or only choose data having high confidence scores?
2. The model you used for the pseudo labels step was fixed?
3. You tried to use unlabeled real-world data for Semi-Supervised Learning on Synthetic data and fall. Can you provide me with some detail about this experiment (setting, hyperparameter, protocol)?
4. Finally, I will be very grateful if you give me some of your comments about the failed experiment mentioned in the question above.

From the bottom of my heart, I'd be grateful if I could get a response from you. It will help my project a lot.

**Comment by KhaLee2307** [comment]
Additionally, one more question, when semi-supervised learning, did you freeze any layer?

Thank you.

**Comment by ku21fan** [comment]
Hi, 
Thank you for your interest in our work :) 

1. We do not choose data having high confidence scores. We do slight filtering here though. After this filtering, we used all of them.
https://github.com/ku21fan/STR-Fewer-Labels/blob/e6aa817e2eacbf29b3fcd11390d78b1a8f96bf78/modules/semi_supervised.py#L54-L60

2. Yes, it is fixed.
3. and 4. Do you mean the fine-tuning experiments in Table 5?
If so, it was not semi-supervised learning on synthetic data. 
For the experiments, after pretraining a model on synthetic data, we fine-tuned the model on real data. 
Thus semi-supervised learning is conducted with real data.

Additional question: No freeze. Although we made the freeze option [here](https://github.com/ku21fan/STR-Fewer-Labels/blob/e6aa817e2eacbf29b3fcd11390d78b1a8f96bf78/train.py#L159-L186), we did not use it for our paper.

Best,
